pkg/controller/notification/notification_delivery_service_test.go

```diff
@@ -20,25 +20,41 @@ func NewMockNotificationDeliveryServiceConfig(service string) NotificationDelive
 	return &MockNotificationDeliveryServiceConfig{service: service}
 }
 
+type MockMailgunConfiguration struct {
+	Domain      string
+	APIKey      string
+	SenderEmail string
+}
+
+func (c *MockMailgunConfiguration) GetMailgunDomain() string {
+	return c.Domain
+}
+
+func (c *MockMailgunConfiguration) GetMailgunAPIKey() string {
+	return c.APIKey
+}
+
+func (c *MockMailgunConfiguration) GetMailgunSenderEmail() string {
+	return c.SenderEmail
+}
+
+func NewMockMailgunConfiguration(domain, apikey, senderEmail string) MailgunConfiguration {
+	return &MockMailgunConfiguration{
+		Domain:      domain,
+		APIKey:      apikey,
+		SenderEmail: senderEmail,
+	}
+}
+
 func TestNotificationDeliveryServiceFactory(t *testing.T) {
 	// given
 	client := test.NewFakeClient(t)
 
-	t.Run("factory configured with mock delivery service", func(t *testing.T) {
-		// when
-		factory := NewNotificationDeliveryServiceFactory(client,
-			NewMockNotificationDeliveryServiceConfig(configuration.NotificationDeliveryServiceMock))
-		svc, err := factory.CreateNotificationDeliveryService()
-
-		// then
-		require.NoError(t, err)
-		require.IsType(t, &MockNotificationDeliveryService{}, svc)
-	})
-
 	t.Run("factory configured with mailgun delivery service", func(t *testing.T) {
 		// when
 		factory := NewNotificationDeliveryServiceFactory(client,
-			NewMockNotificationDeliveryServiceConfig(configuration.NotificationDeliveryServiceMailgun))
+			NewMockNotificationDeliveryServiceConfig(configuration.NotificationDeliveryServiceMailgun),
+			NewMockMailgunConfiguration("foo.com", "12345", "sender@foo.com"))
 		svc, err := factory.CreateNotificationDeliveryService()
```

Comment:
after creating the service, maybe we should have a test that actually calls `Send`?

Reply:
Test now added. :)